homeassistant/components/samsungtv/media_player.py

```diff
@@ -123,127 +140,136 @@ def update(self):
         else:
             self._state = STATE_ON if self._bridge.is_on() else STATE_OFF
 
-    def send_key(self, key):
+    def send_key(self, key: str) -> None:
         """Send a key to the tv and handles exceptions."""
         if self._power_off_in_progress() and key != "KEY_POWEROFF":
             LOGGER.info("TV is powering off, not sending command: %s", key)
             return
         self._bridge.send_key(key)
 
-    def _power_off_in_progress(self):
+    def _power_off_in_progress(self) -> bool:
         return (
             self._end_of_power_off is not None
             and self._end_of_power_off > dt_util.utcnow()
         )
 
     @property
-    def unique_id(self) -> str:
+    def unique_id(self) -> str | None:
         """Return the unique ID of the device."""
         return self._uuid
 
     @property
-    def name(self):
+    def name(self) -> str | None:
         """Return the name of the device."""
         return self._name
 
     @property
-    def state(self):
+    def state(self) -> str | None:
         """Return the state of the device."""
         return self._state
 
     @property
-    def available(self):
+    def available(self) -> bool:
         """Return the availability of the device."""
         if self._auth_failed:
             return False
         return (
             self._state == STATE_ON
-            or self._on_script
-            or self._mac
+            or self._on_script is not None
+            or self._mac is not None
             or self._power_off_in_progress()
         )
 
     @property
-    def device_info(self):
+    def device_info(self) -> DeviceInfo | None:
         """Return device specific attributes."""
-        info = {
-            "name": self.name,
-            "identifiers": {(DOMAIN, self.unique_id)},
-            "manufacturer": self._manufacturer,
-            "model": self._model,
-        }
-        if self._mac:
-            info["connections"] = {(CONNECTION_NETWORK_MAC, self._mac)}
-        return info
+        if (
+            self.name is not None
+            and self._manufacturer is not None
+            and self._model is not None
+            and self.unique_id is not None
+        ):
+            info: DeviceInfo = {
+                "name": self.name,
+                "identifiers": {(DOMAIN, self.unique_id)},
+                "manufacturer": self._manufacturer,
```

Comment:
Avoid including the key in the device info if it's none instead of not returning the whole device info

Reply:
Done

Reply:
Did you forget to push?

Reply:
Was just distracted by a customer call till now ;-)